ArtOfIllusion/src/artofillusion/LayoutWindow.java

```diff
@@ -2184,18 +2225,46 @@ public void transformObjectCommand()
         scaledObjects.add(obj);
       }
     }
-    for (i = 0; i < sel.length; i++)
+    for (int i = 0; i < sel.length; i++)
     {
-      info = theScene.getObject(sel[i]);
-      theScene.objectModified(info.getObject());
+        theScene.objectModified(theScene.getObject(sel[i]).getObject());
     }
-    ArrayList<ObjectInfo> modified = new ArrayList<ObjectInfo>();
+    final ArrayList<ObjectInfo> modified = new ArrayList<ObjectInfo>();
     for (int index : sel)
       modified.add(theScene.getObject(index));
     theScene.applyTracksAfterModification(modified);
     setUndoRecord(undo);
-    updateImage();
   }
+    /** Determine a rotation matrix given the appropriate angles
+     */
+        
+    public static Mat4 determineRotationMatrix(double x, double y, double z) {
+        Mat4 m = Mat4.identity();
+        if (!Double.isNaN(x))
+            m = m.times(Mat4.xrotation(x*Math.PI/180.0));
+        if (!Double.isNaN(y))
+            m = m.times(Mat4.yrotation(y*Math.PI/180.0));
+        if (!Double.isNaN(z))
+            m = m.times(Mat4.zrotation(z*Math.PI/180.0));
+        return m;
+    }
+
+    /** Find the center of all selected objects.
+     */
+    public static Vec3 findSelectedObjectsCenter(Scene scene, int[] sel) {
+        if (sel.length == 0)
+            throw new ArrayIndexOutOfBoundsException("No object selected.");
+        BoundingBox bounds = null;
+        for (int i = 0; i < sel.length; i++)
+        {
+            ObjectInfo info = scene.getObject(sel[i]);
+            if (bounds == null)
+                bounds = info.getBounds().transformAndOutset(info.getCoords().fromLocal());
+            else
+                bounds = bounds.merge(info.getBounds().transformAndOutset(info.getCoords().fromLocal()));
+        }   
+        return bounds.getCenter();
+    }
```

Comment:
Do you see any reason that these might need to be called from outside the class? If not, they should not be public.

Reply:
First, a quick _mea culpa_ : those changes are actually not related to this PR, but I forgot about them when submitting the PR. They should have been in another branch, but when I made them I hadn't created one, and I forgot about these changes when submitting the PR so here they are, out of place. 

Those changes were actually made in correlation with my re-creation (through scripting) of the seven AOI splash screens ("title images"). And those changes are really not additions to the code, the GIT diff got a little lost here : all I did was refactor and isolate those two methods, but all of this code was already present in the `transformObjectCommand()` method. 
As a side note, I totally agree they belong in a math package and another class, and not in this class (which suffers from a huge God-Class syndrome - in other words, it's way too big and does way too many things). 

And now, back to the reasoning about why I made those changes : I feel Scripts and plugins should be able to easily invoke standard AOI Object menu tools (in this case, I needed to invoke the "Object/transform Object" menu item, but the reasoning applies to all of the "Object" menu  items and most of the Tools). I mean, doing things with code instead of by hand_is_ basically the purpose of scripting, isn't it? 

So, if the basic AOI object manipulation menu items are to be invoked by scripting / plugins, the actual "working" and utility code should be smore eparate from the GUI code, and be made public - ideally, in separate classes (`BasicTools` or something equivalent), without any reference to GUI code. 


In the end, I'm sorry this code change appears in this branch and PR, but remember all of it was already present in the class source code, the change is really just some lines moved around. 

What do you think of my reasoning about 

- separating GUI code from "work" code, 
- LayoutWindow doing too many things, and 
- what scripts should be able to do / what should be public?

Reply:
> I forgot about them when submitting the PR. They should have been in another branch

Ahh... That happens. We should figure out where, exactly, they belong. (& what signature) I really don't want to expose them as `public` when we are very likely to be moving them elsewhere.

I'd agree that `LayoutWindow` is very large, and does a lot of things. If some of them are more naturally a separate concern, we should figure out where/how they should be used. But that's another project.

Separation between GUI building and business logic would be welcome. I'm not sure that we need to break them out into separate *classes*, as sometimes the UI itself needs to look at the data closely. (See for example the code for all of the raytracer options. Some of them depend on other options. Would *you* want to figure out how to communicate which parts of the UI are supposed to be disabled through an interface?) Moving logic into a set of methods with a public API - and no implicit dependencies on having a desktop environment available - would definitely be something I would support. Way out of scope for this PR, though.

Yes, I think that scripts should be able to leverage most AOI code, and PRs in support of that are welcome. Do expect some discussion on getting the exposed API right in those cases.

Reply:
It looks like you backed out the changes to the code that *used* these methods, but have left the methods themselves? These should wait until we figure out where they should live.